lib/auth/auth_with_roles.go

```diff
@@ -2624,6 +2655,14 @@ func (a *ServerWithRoles) GetAccessCapabilities(ctx context.Context, req types.A
 		}
 	}
 
+	canFilter, err := a.canFilterRequestableRolesByResource(ctx, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if !canFilter {
+		req.RequestableResourceIDs = nil
+	}
```

Comment:
I don't think ServerWithRoles is the right layer for this, I'd make it an implementation detail of GetAccessCapabilities